src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1376,7 +1383,7 @@ RelationInfoContainsRecurringTuples(PlannerInfo *plannerInfo, RelOptInfo *relati
 		RangeTblEntry *rangeTableEntry = plannerInfo->simple_rte_array[relationId];
 
 		/* relationInfo has this range table entry */
-		if (IsRecurringRTE(rangeTableEntry, recurType))
+		if (ContainsRecurringRTE(rangeTableEntry, recurType))
```

Comment:
@marcocitus you had a comment here, but I cannot see it anymore.

One approach I consider is to get rid of `ShouldRecurseForRecurringTuplesJoinChecks()` and `RelationInfoContainsRecurringTuples`, and only rely on `RelationInfoContainsOnlyRecurringTuples `. But given that we're backporting this change, let's not be invasive. 

Are you OK with that?

Reply:
That change is preferable from a correctness and simplicity standpoint, but we need to study its performance implications since we might recurse too often.